packages/webdriverio/src/commands/element/moveTo.ts

```diff
@@ -30,6 +32,7 @@ export async function moveTo (
      * W3C way of handle the mouse move actions
      */
     const browser = getBrowserObject(this)
+    await moveInViewport(this, scrollTimeout ? scrollTimeout : 2000)
```

Comment:
Let's get rid of moveInViewport, it's doing the same thing as the scrollIntoView command, so we can use that.

Reply:
No exactly the version is 

`export async function isInViewport (elm: WebdriverIO.Element) {
    const browser = getBrowserObject(elm)
    return browser.execute(isElementInViewportScript, {
        [ELEMENT_KEY]: elm.elementId, // w3c compatible
        ELEMENT: elm.elementId // jsonwp compatible
    } as any as HTMLElement)
}
export async function moveInViewport (elm: WebdriverIO.Element, scrollTimeout  = 2000) {
    const browser = getBrowserObject(elm)
    try {
        if (await isInViewport(elm)) {
            await  elm.scrollIntoView({ block : 'nearest', inline: 'nearest', behavior: 'instant' })
            await browser.waitUntil(async ()=> await isInViewport(elm), { timeout: scrollTimeout })
        }
    } catch {
        await  elm.scrollIntoView({ block : 'center', inline: 'center', behavior: 'instant' })
    }
}`

it's checking isInViewport before scrolling, wait for animation in browser.waitUntil and then use scrollIntoView again if it's  still not into view

Reply:
But why do this when scrollIntoView already has that fallback?

Reply:
For the same reason like it's been implemented in isElementClickable here

https://github.com/webdriverio/webdriverio/blob/main/packages/webdriverio/src/scripts/isElementClickable.ts

// Should definitely work even if element was covered with sticky header/footer

Reply:
> But why do this when scrollIntoView already has that fallback?

I use it twice

await  elm.scrollIntoView({ block : 'nearest', inline: 'nearest', behavior: 'instant' })
await  elm.scrollIntoView({ block : 'center', inline: 'center', behavior: 'instant' })

unfortunately i can't figure out where is callback in scrollIntoView with the options.

Reply:
let's get imagine, i'm calling scrollIntoView like
await elm.scrollIntoView({ block : 'nearest', inline: 'nearest', behavior: 'instant' })

after some calculation it's failed here and throw error for example

await browser.action('wheel')
            .scroll({ duration: 0, x: deltaX, deltaY, origin: this })
            .perform()

the it's moved to catch and tried to call scrollIntoView web version but with the same options like { block : 'nearest', inline: 'nearest', behavior: 'instant' }, element still isn't available